File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        q = []
        answers = defaultdict(list)
        
        q.append([root, 0])
        
        while q:
            node, depth = q.pop(0)
            answers[depth].append(node.val)

            if node.left:
                q.append([node.left, depth + 1])
            if node.right:
                q.append([node.right, depth + 1])

        return list(answers.values())

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] answers = new int[nums.length * 2];
        int n = nums.length;

        for(int i = 0; i < n; i++) {
            answers[i] = nums[i];
            answers[i + n] = nums[i];
        }

        return answers;
    }
}